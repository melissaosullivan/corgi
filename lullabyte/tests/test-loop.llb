void main() {
	sound c;
	sound d;
	sound e;
	sound[] scale;
	sound s;
	
	double[] durs;
	double diter;
	int i;
	
	int[] amps;
	int aiter;
	
	pitch[] pits;
	pitch piter;
	
	c = |C1|:0.25:100;
	d = |D1|:0.25:100;
	e = |E1|:0.25:100;
	
	scale =	[c, d, e];
	print(scale);

	loop(s : scale) {
		s = s * 3;
		print(getPitches(s));
		print(getDuration(s));
		print(getAmplitude(s));
	}
	print(scale);

	durs = [0.1, 0.2, 0.3];
	i = 0;
	loop(diter : durs) {
		scale[i] = setDuration(scale[i], diter);
		i = i + 1;
	}
	print(scale);
	
	amps = [25, 50, 75];
	i = 0;
	loop(aiter : amps) {
		scale[i] = setAmplitude(scale[i], aiter);
		i = i + 1;
	}
	print(scale);
	
	pits = [A1, B1, F1];
	i = 0;
	loop(piter : pits) {
		scale[i] = setPitches(scale[i], piter);
		i = i + 1;
	}
	print(scale);
	
}
