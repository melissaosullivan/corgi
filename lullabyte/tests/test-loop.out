[|C1|:0.25:100, |D1|:0.25:100, |E1|:0.25:100]
[C1]
0.75
100
[D1]
0.75
100
[E1]
0.75
100
[|C1|:0.75:100, |D1|:0.75:100, |E1|:0.75:100]
[|C1|:0.1:100, |D1|:0.2:100, |E1|:0.3:100]
[|C1|:0.1:25, |D1|:0.2:50, |E1|:0.3:75]
[|A1|:0.1:25, |B1|:0.2:50, |F1|:0.3:75]
